tests/subsys/pm/power_mgmt/src/dummy_driver.c

```diff
@@ -9,33 +9,88 @@
 #include <pm/device_runtime.h>
 #include "dummy_driver.h"
 
+static bool dont_sleep;
+extern bool z_sys_post_kernel;
+
 static int dummy_open(const struct device *dev)
 {
-	return pm_device_get(dev);
+	dont_sleep = false;
+	pm_device_get(dev);
+	if (dev->pm->enable == false) {
+		pm_device_enable(dev);
+		return pm_device_get(dev);
+
+	}
+	return 0;
 }
 
 static int dummy_close(const struct device *dev)
 {
-	return pm_device_put(dev);
+	dont_sleep = false;
+	pm_device_put(dev);
+	if (z_sys_post_kernel == false) {
+		z_sys_post_kernel = true;
+		return 0;
+	}
+	pm_device_disable(dev);
+	return 0;
+}
+
+static int dummy_refuse_to_sleep(const struct device *dev)
+{
+	dont_sleep = true;
```

Comment:
`bool pm_device_wakeup_enable(struct device *dev, bool enable);`

Reply:
"dont_sleep" is used to break pm_system_suspend() when pm_low_power_devices() or pm_suspend_devices() is called.
pm_device_wakeup_enable() just cause this device is skipped, can't prevent pm_system_suspend() to sleep.